dipy/workflows/vis_registeration.py

```diff
@@ -0,0 +1,278 @@
+        img[..., 0] = np.interp(img[..., 0], (img[..., 0].min(),
+                                              img[..., 0].max()), (0, 14))
+        img[..., 1] = np.interp(img[..., 1], (img[..., 1].min(),
+                                              img[..., 1].max()), (0, 14))
+        img = np.round(img, 0).astype('uint8')
+
+        # Interpolating back to the range 0-255 for getting GIF range.
+        img[..., 0] = np.interp(img[..., 0], (img[..., 0].min(),
+                                              img[..., 0].max()), (0, 255))
+        img[..., 1] = np.interp(img[..., 1], (img[..., 1].min(),
+                                              img[..., 1].max()), (0, 255))
+
+        img = np.round(img, 0).astype('uint8')
+
+        return img
+
+    def animate_overlap_with_renderer(self, static_img, moved_img,
+                                      sli_type, fname, affine):
+
+        """
+        Function for creating the animated GIF from the slices of the
+        registered image.
+        It uses the renderer object to control the dimensions of the
+        created GIF and for correcting the orientation.
+        Patameters
+        ----------
+        sli_type : str (optional)
+            the type of slice to be extracted:
+            sagital, coronal, axial, None (default).
+
+        fname: str, optional
+            Filename for saving the GIF (default 'animation.gif').
+        """
+
+        overlay, value_range = self.process_image_data(static_img,
+                                                       moved_img)
+        x, y, z, _ = overlay.shape
+
+        num_slices = 0
+
+        if sli_type == 'saggital':
+            num_slices = x
+            slice_type = 0
+        elif sli_type == 'coronal':
+            num_slices = y
+            slice_type = 1
+        elif sli_type == 'axial':
+            num_slices = z
+            slice_type = 2
+
+        # Creating the renderer object and setting the background.
+        renderer = window.renderer((0.5, 0.5, 0.5))
+        slices = []
+
+        for i in range(num_slices):
+            temp_slice = overlay_slices(overlay[..., 0], overlay[..., 1],
+                                        slice_type=slice_type,
+                                        slice_index=i, ret_slice=True)
+            temp_slice = temp_slice[..., None]
+            temp_slice = np.round(temp_slice, 0).astype('uint8')
+            temp_slice = np.rollaxis(temp_slice, 2, 4)
+            slice_actor = actor.slicer(temp_slice, affine, value_range)
+            renderer.add(slice_actor)
+            renderer.reset_camera()
+            renderer.zoom(1.6)
+            snap = snapshot(renderer)
+            snap1 = self.adjust_color_range(snap)
+            slices.append(snap1)
+
+        # Writing the GIF below
+        write_gif(slices, fname, fps=10)
+
+    @staticmethod
+    def check_dimensions(static, moving):
+
+        """Check the dimensions of the input images."""
```

Comment:
What's your standard? You sometime glue the documentation to your function, sometime not. I would advise removing the useless \n, but wathever you do, can you please stick to one?